### PR TITLE
Migrate `desktop_photo_search` to top level

### PR DESCRIPTION
**Pull request by domesticmouse**
## Pre-launch Checklist

- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I read the [Contributors Guide].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Contributors Guide]: https://github.com/flutter/samples/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: desktop_photo_search/fluent_ui/pubspec.yaml
Comment (RedBrogdon) [inline desktop_photo_search/fluent_ui/pubspec.yaml:26]: Is this something that'll get published in the future?

File: desktop_photo_search/fluent_ui/pubspec.yaml
Comment (RedBrogdon) [inline desktop_photo_search/fluent_ui/pubspec.yaml:49]: Just checking: is it cool for this GUID to be public?

**Review by RedBrogdon — APPROVED** [review]
LGTM, but hold off on landing this until #1006 lands.

File: desktop_photo_search/fluent_ui/lib/src/widgets/photo_details.dart
Comment (domesticmouse) [reply desktop_photo_search/fluent_ui/lib/src/widgets/photo_details.dart:39]: It's required by the API usage spec 😎

File: desktop_photo_search/fluent_ui/pubspec.yaml
Comment (domesticmouse) [reply desktop_photo_search/fluent_ui/pubspec.yaml:26]: There is ongoing work on design of the API for menus.

File: desktop_photo_search/fluent_ui/pubspec.yaml
Comment (domesticmouse) [reply desktop_photo_search/fluent_ui/pubspec.yaml:49]: yup

File: desktop_photo_search/fluent_ui/lib/src/widgets/policy_dialog.dart
Comment (domesticmouse) [reply desktop_photo_search/fluent_ui/lib/src/widgets/policy_dialog.dart:35]: There should really be an else here with an altert or the like.

**Pull request merged**
